src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/RewindDataIndexRepository.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using DBreeze;
+using DBreeze.DataTypes;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.Features.Consensus.CoinViews;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.ProvenBlockHeaders
+{
+    /// <summary>
+    /// Persistent implementation of the <see cref="RewindData"/> DBreeze repository.
+    /// </summary>
+    public class RewindDataIndexRepository : IRewindDataIndexRepository
+    {
+        /// <summary>
+        /// Instance logger.
+        /// </summary>
+        private readonly ILogger logger;
+
+        /// <summary>
+        /// Access to DBreeze database.
+        /// </summary>
+        private readonly DBreezeEngine dbreeze;
+
+        /// <summary>
+        /// DBreeze table names.
+        /// </summary>
+        private const string RewindDataIndexTable = "RewindDataIndex";
+
+        /// <summary>
+        /// Initializes a new instance of the object.
+        /// </summary>
+        /// <param name="dBreezeCoinView"><see cref="DBreezeCoinView"/>.</param>
+        /// <param name="loggerFactory">Factory to create a logger for this type.</param>
+        public RewindDataIndexRepository(DBreezeCoinView dBreezeCoinView, ILoggerFactory loggerFactory)
+        {
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+
+            this.dbreeze = dBreezeCoinView.DBreeze;
```

Comment:
Instead of exposing this as a public property, I'd create in this class a CreateTransaction method that return a DBreeze Transaction to be used outside. In the future we'd want to use a ITransactionFactory service to inject into constructor of classes that needs a DBreeze Transaction.